### PR TITLE
Access event logging to file

### LINKED ISSUE #73: Log timestamp and source IP addresses
The log should at least include timestamp, source IP address and the URI.

### PR DESCRIPTION
**Pull request by surajt97**
Log of timestamp, source ip and URL in /opt/snare/snare.log

### CODE REVIEW

**Comment by glaslos** [comment]
Please make this optional using a command line flag.

File: snare.py [hunk 1]
```diff
@@ -178,6 +179,10 @@ def handle_request(self, request, payload):
         # Submit the event to the TANNER service
         event_result = yield from self.submit_data(data)
 
+        #Log the URL to /opt/snare/snare.log
+        ip=self.transport.get_extra_info('peername')[0]
+        self.logger.info('Source IP %s - - Requested path %s', ip, self.run_args.page_dir + request.path)
```
Comment (glaslos) [inline snare.py]: Please use the Apache access log format: https://httpd.apache.org/docs/current/logs.html#accesslog

File: snare.py [hunk 2]
```diff
@@ -461,6 +466,11 @@ def check_tanner_connection():
         args.interface, int(args.port))
     srv = loop.run_until_complete(future)
 
+    #log info
+    info_log_file_name = '/opt/snare/snare.log'
+    logger.Logger.create_logger(info_log_file_name, __package__)
```
Comment (glaslos) [inline snare.py]: Pass the logger to `HttpRequestHandler` as argument.

File: logger.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    def filter(self, record):
+        return record.levelno < self.level # "<" instead of "<=": since logger.setLevel is inclusive, this should be exclusive
+
+
+class Logger:
+    @staticmethod
+    def create_logger(debug_filename, logger_name):
+        logger = logging.getLogger(logger_name)
+        logger.setLevel(logging.INFO)
```
Comment (glaslos) [inline logger.py]: Expose the log level as command line flag

File: logger.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+        # DEBUG log to '/opt/snare/snare.log'
+        debug_log_handler = logging.handlers.RotatingFileHandler(debug_filename, encoding='utf-8')
+        debug_log_handler.setLevel(logging.INFO)
+        debug_log_handler.setFormatter(formatter)
+        max_level_filter = LevelFilter(logging.ERROR)
+        debug_log_handler.addFilter(max_level_filter)
+        logger.addHandler(debug_log_handler)
+
+        return logger
```
Comment (glaslos) [inline logger.py]: missing newline

File: snare.py [hunk 5]
```diff
@@ -178,6 +179,10 @@ def handle_request(self, request, payload):
         # Submit the event to the TANNER service
         event_result = yield from self.submit_data(data)
 
+        #Log the URL to /opt/snare/snare.log
+        ip=self.transport.get_extra_info('peername')[0]
```
Comment (glaslos) [inline snare.py]: PEP-8

**Comment by glaslos** [comment]
@surajt97 I consider the formatting of the messages as critical. If we stick to the same format as Apache, people can use other existing access log tools to analyze the Snare logs.

**Comment by surajt97** [comment]
@glaslos should I use aiohttp.access logger?

**Comment by glaslos** [comment]
Yes, that could be an option. Details can be found here: https://docs.aiohttp.org/en/stable/logging.html
You can also just change the format of your current logger. Using aiohttp might require fixing this issue #55 and using something more high level.

**Comment by glaslos** [comment]
@afeena do you have an opinion on this?

**Comment by afeena** [comment]
I'd prefer using aiohttp logging, it can be done even without high-level changes, but solution without aiohttp logging is also ok for me:)

File: snare.py [hunk 6]
```diff
@@ -251,6 +256,19 @@ def handle_request(self, request, payload):
...
+        req = request.method + ' /' + self.run_args.page_dir + request.path
+        major = response.version.major
+        minor = response.version.minor
+        ver = " HTTP/" + str(major) + '.' + str(minor)
+        req = req + ver
+        user = getpass.getuser()
+        tim = strftime("%d/%b/%Y:%H:%M:%S %z")
+        d = {'hostIP': request.headers['Host'], 'stat': response.status, 'req': req, 'user': user,
+             'content_length': response.headers['Content-Length'], 'time': tim}
+        self.logger.info(' ', extra=d)
```
Comment (afeena) [inline snare.py]: I think this part with logging should executed only if we have `logging==True`. In other case this line will be executed with default self.logger of aiohttp server

Comment (surajt97) [reply snare.py]: @afeena pls review.

**Comment by afeena** [comment]
@surajt97  Could you please also take a look at these comments? https://github.com/mushorg/snare/pull/76#discussion_r153746461

**Comment by glaslos** [comment]
Needs a rebase

**Comment by mzfr** [comment]
@afeena I think we should close this since this issue is fixed after merging #109

**Pull request closed without merging**